assets/dev/js/admin/new-template/template-controls.js

```diff
@@ -0,0 +1,45 @@
+export default class TemplateControls {
+	setDynamicFieldsVisibility( lookupControlIdPrefix, controls ) {
+		if ( undefined === controls ) {
+			return;
+		}
+
+		const controlsArray = Object.entries( controls );
+
+		for ( const [ controlId, controlSettings ] of controlsArray ) {
+			this.setVisibilityForControl( lookupControlIdPrefix, controlSettings, controlId );
+		}
+	}
+
+	setVisibilityForControl( lookupControlIdPrefix, controlSettings, controlId ) {
+		const conditions = Object.entries( controlSettings.conditions ?? {} );
+
+		conditions.forEach( ( condition ) => {
+			this.changeVisibilityBasedOnCondition( lookupControlIdPrefix, condition, controlId );
+		} );
+	}
+
+	changeVisibilityBasedOnCondition( lookupControlIdPrefix, condition, controlId ) {
+		const [ conditionKey, conditionValue ] = condition;
```

Comment:
Refactor with `toggleClass` and `force` parameter

Reply:
Done